File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {

        if(nums.length == 0) return false;

        HashSet<Integer> set = new HashSet<>();

        for(int num : nums){
            if(set.contains(num)) 
                return true;
            set.add(num);
            
        }
        return false;
 
    }
}

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        
        result = []
        perm = []
        used = set()
        
        def backtrack():
            if len(perm) == len(nums):
                result.append(perm.copy())
                return
            
            for num in nums:
                if num not in perm:
                    perm.append(num)
                    backtrack()
                    perm.pop()
        
        backtrack()
        return result